### PR TITLE
Added support for comments on tables and columns. Added test cases

### PR DESCRIPTION
**Pull request by ajay-abrol2**

### CODE REVIEW

File: dbt/include/vertica/macros/adapters/persist_docs.sql [hunk 1]
```diff
@@ -1,14 +1,47 @@
-{% macro vertica__alter_column_comment(relation, column_dict) -%}
-  {{ exceptions.raise_not_implemented(
-    'alter_column_comment macro not implemented for adapter '+adapter.type()) }}
+{#
+  Code extracted from https://github.com/dbt-labs/dbt-adapters/blob/61801e9aa22c6bf219398b7fee9eefca96a85e7a/dbt-postgres/src/dbt/include/postgres/macros/adapters.sql#L197
```
Comment (mkottakota1) [inline dbt/include/vertica/macros/adapters/persist_docs.sql:2]: Instead of "extracted by" use "Inspired by"

File: dbt/include/vertica/macros/adapters/persist_docs.sql [hunk 2]
```diff
@@ -1,14 +1,47 @@
...
 
+{% macro vertica__alter_relation_comment(relation, comment) %}
+  {% set escaped_comment = vertica_escape_comment(comment) %}
+  {% if relation.type == 'materialized_view' -%}
+    {% set relation_type = "materialized view" %}
+  {%- else -%}
+    {%- set relation_type = relation.type -%}
+  {%- endif -%}
+  comment on {{ relation_type }} {{ relation }} is {{ escaped_comment }};
+{% endmacro %}
```
Comment (mkottakota1) [inline dbt/include/vertica/macros/adapters/persist_docs.sql:44]: Issue 2: Null Comments Not Handled (Lines 36-43)
Same problem in vertica__alter_relation_comment:

{% macro vertica__alter_relation_comment(relation, comment) %}
  {% set escaped_comment = vertica_escape_comment(comment) %}  {# ← Line 37: Can receive null #}

If a model has no description, comment = null → dbt run fails

Fix needed (lines 36-43):

{% macro vertica__alter_relation_comment(relation, comment) %}
  {% if comment %}  {# Skip null/empty descriptions #}
    {% set escaped_comment = vertica_escape_comment(comment) %}
    {% if relation.type == 'materialized_view' -%}
      {% set relation_type = "materialized view" %}
    {%- else -%}
      {%- set relation_type = relation.type -%}
    {%- endif -%}
    comment on {{ relation_type }} {{ relation }} is {{ escaped_comment }};
  {% endif %}
{% endmacro %}

File: dbt/include/vertica/macros/adapters/persist_docs.sql [hunk 3]
```diff
@@ -1,14 +1,47 @@
...
 
-{% macro vertica__alter_relation_comment(relation, relation_comment) -%}
-  {{ exceptions.raise_not_implemented(
-    'alter_relation_comment macro not implemented for adapter '+adapter.type()) }}
+{% macro vertica__alter_column_comment(relation, column_dict) %}
+  {% if relation.type == 'view' -%}
+    {{ exceptions.warn("[WARNING]: Vertica does not support comments for view columns in view: " ~ relation.name) }}
+  {%- else -%}
+    {% set existing_columns = adapter.get_columns_in_relation(relation) | map(attribute="name") | list %}
+    {% for column_name in column_dict if (column_name in existing_columns) %}
```
Comment (mkottakota1) [inline dbt/include/vertica/macros/adapters/persist_docs.sql:28]: ## FILE: dbt/include/vertica/macros/adapters/persist_docs.sql

### 🔴 CRITICAL BUG #1: Null Column Comments Crash (Lines 28-31)

**Problem:**
```jinja2
{% for column_name in column_dict if (column_name in existing_columns) %}
  {% set comment = column_dict[column_name]['description'] %}     {# ← LINE 29: Can be NULL #}
  {% set escaped_comment = vertica_escape_comment(comment) %}    {# ← LINE 30: CRASHES if null #}
  comment on column {{ relation }}...                            {# ← LINE 31 #}
{% endfor %}

If a user doesn't provide a description for a column in schema.yml:

columns:
  - name: id
    # No description!


Then comment = null and line 30 crashes with: "cannot escape a non-string: null"

Example that fails:

Code
ERROR: cannot escape a non-string: null
  in macro vertica_escape_comment (dbt/include/vertica/macros/adapters/persist_docs.sql)
  caused by dbt.exceptions.DbtCompilerError

FIX - Replace lines 28-31 with:

{% for column_name in column_dict if (column_name in existing_columns) %}
  {% set comment = column_dict[column_name]['description'] %}
  {% if comment %}  {# Skip if description is null or empty #}
    {% set escaped_comment = vertica_escape_comment(comment) %}
    comment on column {{ relation }}.{{ adapter.quote(column_name) if column_dict[column_name]['quote'] else column_name }} is {{ escaped_comment }};
  {% endif %}
{% endfor %}

File: dbt/include/vertica/macros/adapters/persist_docs.sql [hunk 4]
```diff
@@ -1,14 +1,47 @@
...
+      {% set comment = column_dict[column_name]['description'] %}
+      {% set escaped_comment = vertica_escape_comment(comment) %}
+      comment on column {{ relation }}.{{ adapter.quote(column_name) if column_dict[column_name]['quote'] else column_name }} is {{ escaped_comment }};
+    {% endfor %}
+  {%- endif -%}
 {%- endmacro %}
 
+{% macro vertica__alter_relation_comment(relation, comment) %}
+  {% set escaped_comment = vertica_escape_comment(comment) %}
```
Comment (mkottakota1) [inline dbt/include/vertica/macros/adapters/persist_docs.sql:37]: Problem:

{% macro vertica__alter_relation_comment(relation, comment) %}
  {% set escaped_comment = vertica_escape_comment(comment) %}  {# ← LINE 37: Can be NULL #}

If a model has no description:

models:
  - name: transactions
    # No description!

Same crash: "cannot escape a non-string: null"

FIX - Wrap lines 37-43 with:

{% macro vertica__alter_relation_comment(relation, comment) %}
  {% if comment %}  {# Skip if description is null or empty #}
    {% set escaped_comment = vertica_escape_comment(comment) %}
    {% if relation.type == 'materialized_view' -%}
      {% set relation_type = "materialized view" %}
    {%- else -%}
      {%- set relation_type = relation.type -%}
    {%- endif -%}
    comment on {{ relation_type }} {{ relation }} is {{ escaped_comment }};
  {% endif %}
{% endmacro %}

File: dbt/include/vertica/macros/adapters/persist_docs.sql [hunk 5]
```diff
@@ -1,14 +1,47 @@
-{% macro vertica__alter_column_comment(relation, column_dict) -%}
-  {{ exceptions.raise_not_implemented(
-    'alter_column_comment macro not implemented for adapter '+adapter.type()) }}
+{#
+  Code extracted from https://github.com/dbt-labs/dbt-adapters/blob/61801e9aa22c6bf219398b7fee9eefca96a85e7a/dbt-postgres/src/dbt/include/postgres/macros/adapters.sql#L197
+#}
+
+
+{#
+  By using dollar-quoting like this, users can embed anything they want into their comments
```
Comment (mkottakota1) [inline dbt/include/vertica/macros/adapters/persist_docs.sql:7]: This is only visible to developers. Users won't see this limitation.

Suggestion: Add to PR description or README:

## Known Limitations for Table & Column Comments

- Comments **cannot contain** the exact string: `$dbt_comment_literal_block$`
  - This is an internal PostgreSQL dollar-quoting delimiter used for SQL safety
  - If this string appears in a description, dbt will raise:
    ```
    The string $dbt_comment_literal_block$ is not allowed in comments.
    ```
- **Workaround:** Slightly rephrase your comment to avoid this string (very unlikely edge case)

**Example that fails:**
```yaml
models:
  - name: my_table
    description: "Uses $dbt_comment_literal_block$ for escaping"
    # This will fail!


What works instead:

models:
  - name: my_table
    description: "Uses dollar-quoting delimiters for escaping"
    # This will work

File: tests/functional/adapter/test_persist_docs.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        results, log_output = run_dbt_and_capture(["--debug", "run"])
+
+        assert len(results) == 1
+
+        normalized_log = log_output.lower()
+        assert "comment on table" in normalized_log
+        assert "comment on column" in normalized_log
+        assert "table comment from dbt" in normalized_log
+        assert "id column comment from dbt" in normalized_log
+        assert "first_name column comment from dbt" in normalized_log
```
Comment (mkottakota1) [inline tests/functional/adapter/test_persist_docs.py:45]: Current test (lines 35-45) only covers the happy path where descriptions exist.
Add edge case test cases to validate 

regressions with null descriptions
Validates materialized view support
Confirms special character escaping works

File: tests/conftest.py [hunk 7]
```diff
@@ -35,51 +32,10 @@ def pytest_configure(config):
...
-#         'password': '',
-#         'database': 'docker',
-#         'port': 5433,
-#     }
-
 @pytest.fixture(scope="session")
 def dbt_profile_target(request):
     profile_type = request.config.getoption("--profile")
-    if profile_type =="vertica":
+    if profile_type == "vertica":
```
Comment (mkottakota1) [inline tests/conftest.py:38]: nit: Fix the indentation.

File: tests/conftest.py [hunk 8]
```diff
@@ -89,6 +45,7 @@ def dbt_profile_target(request):
         raise ValueError(f"Invalid profile type '{profile_type}'")
     return target
 
+
```
Comment (mkottakota1) [inline tests/conftest.py:48]: nit:remove the extra space/new ln

File: tests/conftest.py [hunk 9]
```diff
@@ -101,15 +58,17 @@ def databricks_sql_endpoint_target():
         "host": "localhost",
     }
 
+
```
Comment (mkottakota1) [inline tests/conftest.py:61]: remove the extra space/new ln

File: tests/conftest.py [hunk 10]
```diff
@@ -101,15 +58,17 @@ def databricks_sql_endpoint_target():
...
 def vertica_target():
-    return {
-       'type': 'vertica',
+   return {
+         'type': 'vertica',
```
Comment (mkottakota1) [inline tests/conftest.py:64]: remove the extra space/new ln

or check the indentation

File: tests/conftest.py [hunk 11]
```diff
@@ -101,15 +58,17 @@ def databricks_sql_endpoint_target():
...
         'threads': 1,
         'host': 'localhost',
         'username': 'dbadmin',
         'password': '',
         'database': 'docker',
         'port': 5433,
+
```
Comment (mkottakota1) [inline tests/conftest.py:71]: remove the extra space/new ln

File: tests/functional/adapter/test_grants.py [hunk 12]
```diff
@@ -48,7 +48,7 @@
     config:
       materialized: table
       grants:
-        fake_privilege: ["{{ env_var('DBT_TEST_USER_2') }}"]
+                fake_privilege: ["{{ env_var('DBT_TEST_USER_2', 'invalid_user') }}"]
```
Comment (mkottakota1) [inline tests/functional/adapter/test_grants.py:51]: nit: check the indentation

**Review by mkottakota1 — CHANGES_REQUESTED** [review]
